src/components/analytics/services/znkAnalyticsUtilSrv.js

```diff
@@ -0,0 +1,89 @@
+(function (angular) {
+    'use strict';
+
+    function _getTimeInDay() {
+        var date = new Date();
+        var hours = date.getHours();
+        var timeStr;
+
+        if(hours >= 6 && hours < 12) {
+            timeStr = 'Morning';
+        } else if(hours >= 12 && hours < 18) {
+            timeStr = 'Afternoon';
+        } else if(hours >= 18 && hours < 24) {
+            timeStr = 'Evening';
+        } else if(hours >= 24 && hours < 6) {
+            timeStr = 'Night';
+        } else {
+            timeStr = date.toString();
+        }
+
+        return timeStr;
+    }
+
+    function _getQuestionsStats(arr) {
+        return arr.reduce(function(previousValue, currentValue) {
+            if(currentValue.userAnswer) {
+                if(currentValue.isAnsweredCorrectly) {
+                    previousValue.correct++;
+                } else {
+                    previousValue.wrong++;
+                }
+            } else {
+                previousValue.skip++;
+            }
+            return previousValue;
+        },{ correct: 0, wrong: 0, skip: 0 });
+    }
+
+    angular.module('znk.infra.analytics').service('znkAnalyticsUtilSrv', ['$log', function ($log) {
```

Comment:
the znkAnalyticsUtilSrv is in a difrrent file from znkAnalyticsSrv, but this helper function are part of the znkAnalyticsUtilSrv